src/App.tsx

```diff
@@ -1,14 +1,56 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [input, setInput] = useState('');
+  const [isInputFocused, setInputFocused] = useState(false);
+  const [appliedInput, setAppliedInput] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState(peopleFromServer[0]);
+  const [isSelected, setIsSelected] = useState(false);
+  const { name, born, died } = selectedPerson;
+
+  const timerId = useRef(0);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => person
+      .name.toLowerCase().includes(appliedInput.toLowerCase()));
+  }, [appliedInput]);
+
+  function debounce(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
it's bad practice to use different approaches among the project, if u use arrow functions everywhere, make ur debounce an arrow func as well